crypto/evp/m_sigver.c

```diff
@@ -127,6 +181,28 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
         goto legacy;
     }
 
+    /*
+     * Ensure that the key is provided, either natively, or as a cached export.
+     */
+    provkey = evp_pkey_export_to_provider(locpctx->pkey, locpctx->libctx,
+                                          &tmp_keymgmt, locpctx->propquery);
+    if (provkey == NULL) {
+        /*
+         * We assume that the pkey's key management does not support export.
+         * Now we need to see if we can fallback to using an operation from the
+         * same provider as the key.
+         */
+        EVP_SIGNATURE_free(signature);
+
+        ret = evp_handle_unexportable_key_for_sig(locpctx, supported_sig,
+                                                  &tmp_keymgmt, &signature,
+                                                  &provkey);
```

Comment:
This causes a segfault because tmp_keymgmt is NULL at this point. Looks like its nulled by the `...export_to_provider` call when it fails.

Adding `tmp_keymgmt = locpctx->keymgmt;` just above this fixes that.